aptos-move/framework/aptos-framework/sources/Token.move

```diff
@@ -294,14 +312,16 @@ module AptosFramework::Token {
     }
 
     public fun initialize_token_store(account: &signer) {
-        move_to(
-            account,
-            TokenStore {
-                tokens: Table::new(),
-                deposit_events: Event::new_event_handle<DepositEvent>(account),
-                withdraw_events: Event::new_event_handle<WithdrawEvent>(account),
-            },
-        );
+        if (!exists<TokenStore>(Signer::address_of(account))) {
```

Comment:
Why add this check? There's already a similar check where initialize_token_store is called.

Reply:
user can directly call "    public(script) fun initialize_token_script(account: &signer) {". It is also better to let this TXN succeeding without giving them a confusing error message.

Reply:
In that case let's remove the other check then.

Reply:
I'm referring to this duplicate check: https://github.com/aptos-labs/aptos-core/blob/24edd7fbe95773825253a522d1beded80de13206/aptos-move/framework/aptos-framework/sources/Token.move#L250. It's just calling initialize_token which does the same check.

Reply:
removed